sql: TestShowLastQueryStatistics failed

sql.TestShowLastQueryStatistics [failed](https://mesolite.cluster.engflow.com/invocations/default/9f558e5f-9b10-4cf8-8dae-89a7f2469910?testReportRun=2&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [2e2f525a3e650922a18094f322dafbcceb36fc2d](https://github.com/cockroachdb/cockroach/commits/2e2f525a3e650922a18094f322dafbcceb36fc2d):


```
=== RUN   TestShowLastQueryStatistics
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestShowLastQueryStatistics3251297810
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    conn_executor_test.go:1499: unexpected plan latency: 5.199609
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestShowLastQueryStatistics3251297810
--- FAIL: TestShowLastQueryStatistics (14.55s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowLastQueryStatistics.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-56380

--- a/pkg/sql/conn_executor_test.go
+++ b/pkg/sql/conn_executor_test.go
@@ -1408,8 +1408,15 @@ func TestShowLastQueryStatistics(t *testing.T) {
 	skip.UnderRace(t, "measure planning latency which is slower than usual under race")
 
 	ctx := context.Background()
-	params := base.TestServerArgs{}
-	s, sqlConn, _ := serverutils.StartServer(t, params)
+	var testTenant base.DefaultTestTenantOptions
+	if syncutil.DeadlockEnabled {
+		// Under deadlock, the planning latency in the external process mode can
+		// sometimes exceed 1s allowed limit.
+		testTenant = base.TestSkipForExternalProcessMode()
+	}
+	s, sqlConn, _ := serverutils.StartServer(t, base.TestServerArgs{
+		DefaultTestTenant: testTenant,
+	})
 	defer s.Stopper().Stop(ctx)
 
 	testCases := []struct {

